preinstall.sh

```
@@ -0,0 +1,53 @@
+  echo "Detected Linux. Using apt package manager."
+  install_package_linux jq
+  install_package_linux bc
+elif [[ "$OS" == "Darwin" ]]; then
+  echo "Detected macOS. Using Homebrew package manager."
+
+  # Check if Homebrew is installed
+  if ! command -v brew &> /dev/null; then
+    echo "Homebrew is not installed. Installing Homebrew..."
+    /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
+    # Add Homebrew to PATH for immediate use
+    eval "$($(brew --prefix)/bin/brew shellenv)"
```

Shellcheck (0.10.0)

[warning] Quote this to prevent word splitting.

(SC2046)